packages/common/restore.yml

```
@@ -499,9 +513,106 @@ actions:
           echo "The Elasticsearch snapshot you are trying to restore is not complete, aborting" >&2
           exit 1
         fi
+
+        export PAPI_TOKEN="${globals.papiToken}"
+        export PAPI_HOSTNAME="${globals.papiHostname}"
+        export PAPI_ENV_ID="${globals.papiEnvId}"
+        export PAPI_API_VERSION="${globals.papiApiVersion}"
+
+        environment=$(papi.py -X GET "paas-environment/$PAPI_ENV_ID")
+        ec_deployment_id=$(echo $environment | jq -r .ec_deployment_id)
+        ec_deployment=$(papi.py -X GET "ec-deployment/$ec_deployment_id")
+        ec_deployment_mutualized=$(echo $ec_deployment | jq -r .mutualized)
+
+        dev_dedicated=50
+        dev_mutualized=70
+        preprod_dedicated=50
+        preprod_mutualized=70
+        prod_dedicated=20
+        prod_mutualized=50
+
+        if [ "$ec_deployment_mutualized" = true ]; then
+          eval ratio=\$${JEL_ENV_ROLE}_mutualized
+        else
+          eval ratio=\$${JEL_ENV_ROLE}_dedicated
+        fi
+
+        # Get the total shards and score before the restore
+        current_total_shards=$(curl -sS -u $ec_admin_credentials https://$UNOMI_ELASTICSEARCH_ADDRESSES/_cluster/stats | jq -r '.indices.shards.total')
+        current_score=$(( (current_total_shards/ratio) + ((current_total_shards%ratio) > 0 )))
```

Is this really working ?

you should better use bash _indirect reference_ here:
```suggestion
        if $ec_deployment_mutualized; then
          ratio=${JEL_ENV_ROLE}_mutualized
        else
          ratio=$${JEL_ENV_ROLE}_dedicated
        fi

        # Get the total shards and score before the restore
        current_total_shards=$(curl -sS -u $ec_admin_credentials https://$UNOMI_ELASTICSEARCH_ADDRESSES/_cluster/stats | jq -r '.indices.shards.total')
        current_score=$(( (current_total_shards/${!ratio}) + ((current_total_shards%${!ratio}) > 0 )))
```

---

This actually works as confirmed.